optuna/samplers/_nsga2/sampler.py

```diff
@@ -154,18 +162,36 @@ def sample_relative(
         generation = parent_generation + 1
         study._storage.set_trial_system_attr(trial_id, _GENERATION_KEY, generation)
 
+        dominates = _dominates if self._constraints_func is None else _constrained_dominates
         if parent_generation >= 0:
-            p0 = self._select_parent(study, parent_population)
+            # crossover
             if self._rng.rand() < self._crossover_prob:
-                p1 = self._select_parent(
-                    study, [t for t in parent_population if t._trial_id != p0._trial_id]
+                child = crossover(
+                    self._crossover_name,
+                    study,
+                    parent_population,
+                    search_space,
+                    self._rng,
+                    self._swapping_prob,
+                    dominates,
                 )
             else:
-                p1 = p0
+                child = parent_population[0].params
```

Comment:
I'm worried that this new logic could decrease the diversity of children when crossover doesn't occur.
(To the best of my knowledge, `_collect_parent_population` is deterministic and the individuals in `parent_population` and the order of them are the same during the same generation. Thus, `child` seems to always have the same params during the same duration if the crossover is skipped. I'm not 100% sure though)

Could you explain the reason that the first individual in `parent_population` is always selected here?

Reply:
I forgot that `_collect_parent_population` is deterministic.
Do the following.
```suggestion
                parent_population_size = len(parent_population)
                child = parent_population[self._rng.choice(parent_population_size)].params
```